app/src/main/java/com/github/giommok/softwaredevproject/FirebaseAccount.java

```diff
@@ -93,97 +115,160 @@ public String getUsername() {
-                    Log.e("FireBase Data","Can't synchronize user discovered country high points");
-                }
-            }
-            @Override
-            public void onCancelled(@NonNull DatabaseError error) {
-
-            }
-        });
+    public void setDiscoveredPeaks(ArrayList<POIPoint> newDiscoveredPeaks){
+        Database.setChildObjectList(Database.CHILD_USERS + getId() +
+                Database.CHILD_DISCOVERED_PEAKS,
+                new ArrayList<Object>(newDiscoveredPeaks));
+
+    }
+
+    @Override
+    public HashSet<POIPoint> getDiscoveredPeaks(){
+        return discoveredPeaks;
+    }
+
+    /**
+     * Filter the list of discovered peaks. If the peak in the list is already contained in the Hashset (so contained in the DB)
+     * the peak gets dropped from the list.
+     * Updates the HashSet with the new values
+     * @param unfilteredDiscoveredPeaks unfiltered list of peaks (coming directly from AR activity)
+     * @return list of POI after filtering
+     */
+    public ArrayList<POIPoint> filterNewDiscoveredPeaks(ArrayList<POIPoint> unfilteredDiscoveredPeaks){
+        ArrayList<POIPoint> resultList = unfilteredDiscoveredPeaks.stream().filter(newPeak -> !discoveredPeaks.contains(newPeak)).collect(Collectors.toCollection(ArrayList::new));
+        //Update the list of discovered peaks (local HashSet)
+        discoveredPeaks.addAll(resultList);
+        return resultList;
+    }
+
+
+    @Override
+    public void synchronizeUserProfile(){
+        dbRefUser.removeEventListener(userListener);
+        dbRefUser = Database.refRoot.child(Database.CHILD_USERS+  getId());
+        dbRefUser.addValueEventListener(userListener);
+        Log.d("FirebaseAccount","User profile callback created successfully");
+    }
+
+
+    /**
+     * Get the list of discovered peak from the datasnapshot from database.
+     * The list is automatically appended to the discoveredPeaks HashSet
+     * @param discoveredPeaksFromHashMap snapshot from the database user child
+     */
+    private static void syncGetDiscoveredPeaksFromProfile(DataSnapshot discoveredPeaksFromHashMap){
+        HashMap<String, HashMap<String, String>> entries = (HashMap<String, HashMap<String, String>>) discoveredPeaksFromHashMap.getValue();
+
+        for (Map.Entry<String, HashMap<String, String>> entry : entries.entrySet()) {
+            Object value = entry.getValue();
+            POIPoint peak = new POIPoint(((HashMap<String, String>) value).get(Database.CHILD_ATTRIBUTE_PEAK_NAME),
+                    ((HashMap<String, Double>) value).get(Database.CHILD_ATTRIBUTE_PEAK_LATITUDE),
+                    ((HashMap<String, Double>) value).get(Database.CHILD_ATTRIBUTE_PEAK_LONGITUDE),
+                    ((HashMap<String, Long>) value).get(Database.CHILD_ATTRIBUTE_PEAK_ALTITUDE));
+
+            //Check if the peak is already contain to avoid duplicate creation
+            if(!discoveredPeaks.contains(peak)) discoveredPeaks.add(peak);
```

Comment:
HashSet.add() already checks if the object is already contained thus no need to check.
```suggestion
            discoveredPeaks.add(peak);
```